### PR TITLE
airsonic: Add virtualHost option to set up nginx virtual host

### PR DESCRIPTION
**Pull request by telotortium**
Modeled after nixos/modules/services/web-apps/tt-rss.nix. The setup is
slightly non-intuitive, so I think it's worth adding upstream.

###### Things done

- [ ] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS)
- Built on platform(s)
   - [X] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nox --run "nox-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [X] Assured whether relevant documentation is up to date
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

### CODE REVIEW

File: nixos/modules/services/misc/airsonic.nix [hunk 1]
```diff
@@ -25,6 +25,14 @@ in {
         '';
       };
 
+      virtualHost = mkOption {
+        type = types.nullOr types.str;
+        default = "airsonic";
```
Comment (fpletz) [inline nixos/modules/services/misc/airsonic.nix]: The default should be `null`, right?

File: nixos/modules/services/misc/airsonic.nix [hunk 2]
```diff
@@ -116,6 +124,9 @@ in {
           -Dserver.port=${toString cfg.port} \
           -Dairsonic.contextPath=${cfg.contextPath} \
           -Djava.awt.headless=true \
+          ${if (cfg.virtualHost != null)
+            then "-Dserver.use-forward-headers=true"
+            else ""} \
```
Comment (fpletz) [inline nixos/modules/services/misc/airsonic.nix]: You can use `optionalString` instead of `if ... then ... else ""` here.

**Review by fpletz — CHANGES_REQUESTED** [review]

File: nixos/modules/services/misc/airsonic.nix (hunk 1 above)
Comment (telotortium) [reply nixos/modules/services/misc/airsonic.nix]: tt-rss uses a default of `"tt-rss"`. Having said that, `null` is probably a better default choice.

File: nixos/modules/services/misc/airsonic.nix (hunk 2 above)
Comment (telotortium) [reply nixos/modules/services/misc/airsonic.nix]: Done

**Comment by telotortium** [comment]
Hi, could you please take another look at this?

**Review by fpletz — APPROVED** [review]

**Comment by fpletz** [comment]
Thanks!

**Pull request merged**